### PR TITLE
Feature/#99 public profile

### LINKED ISSUE #99: Public Profile
### Description

Implement a public profile in the serviceEndpoint section of the DDO.

### TODO
- [ ] JSON-LD PublicProfile, IPFS file
- [x] check available predicates on schema.org
- [ ] check keyword 'link' on IPLD
- [x] check resolution of arrays on IPLD

### PR DESCRIPTION
**Pull request by charleenfei**
## Description

Fixes #99

> What does it solve? 

Introduces a serviceEndpoints section in the DDO, makes relevant changes to the credentials to enable a PublicProfileCredential.

> Were there any particular challenges that you faced and/or things that were not straightforward? 

## Checklist
- [ ] Keep flags added to the PR up to date
- [ ] Make sure the title describes the code well, description should contain `Fixes #number_of_an_issue` and a brief description of changes
- [ ] Make sure the PR is readable and contains changes only related to a single issue. Otherwise - divide into separate PRs
- [ ] Add new tests and make sure all tests are passing
- [ ] Make sure all edge cases are taken into account and covered by tests
- [ ] Run linter and fix all the errors and warnings

### CODE REVIEW

File: ts/identity/didDocument/types.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
...
 
 export interface IDidDocumentAttrs {
   '@context': string
   id: string
-  authentication: IAuthenticationSectionAttrs[]
-  publicKey: IPublicKeySectionAttrs[]
+  authentication: IAuthenticationSectionAttrs
+  publicKey: IPublicKeySectionAttrs
+  service: IServiceEndpointSectionAttrs
   created: Date
```
Comment (nataschaberg) [inline ts/identity/didDocument/types.ts]: Should the Array indication on type really be removed? authentication, publicKey and service are Arrays on DDO if I am not mistaken.

File: ts/ipfs/index.ts [hunk 2]
```diff
@@ -41,4 +41,26 @@ export class IpfsStorageAgent implements IIpfsConnector {
       throw new Error(`Removing pinned hash ${hash} failed, status code: ${res.status}`)
     }
   }
+
+  public async createDagObject(data: object, pin: boolean): Promise<string> {
+    if (!data || typeof data !== 'object') {
```
Comment (nataschaberg) [inline ts/ipfs/index.ts]: I think we should use args destructuring here in order to be consistent with our current appoach.

File: tests/data/ipfs/ipfs.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export const localHostStorage = 'http://localhost:5001'
```
[the remark this answers is not in the document]
Comment (katszwn) [reply tests/data/ipfs/ipfs.ts]: I think for this case we don't need nesting - unless more ipfs related modules appear. But for general tests - we should have some kind of nesting, at least top level.

File: ts/identity/index.ts [hunk 4]
```diff
@@ -33,18 +29,18 @@ export class Identity {
...
-    const masterKeyPair = keyDerivation.deriveMasterKeyPairFromMnemonic(mnemonic)
-    const genericSigningKey = keyDerivation.deriveGenericSigningKeyPair(masterKeyPair)
-    const ethereumKey = keyDerivation.deriveEthereumKeyPair(masterKeyPair)
-    const ddo = new DidDocument().fromPublicKey(genericSigningKey.publicKey)
+  // public create(randomStringFromEntropy: Buffer): IdentityCreationResponse {
+  //   const mnemonic = keyDerivation.generateMnemonic(randomStringFromEntropy)
+  //   const masterKeyPair = keyDerivation.deriveMasterKeyPairFromMnemonic(mnemonic)
+  //   // const genericSigningKey = keyDerivation.deriveGenericSigningKeyPair(masterKeyPair)
+  //   // const ethereumKey = keyDerivation.deriveEthereumKeyPair(masterKeyPair)
+  //   const ddo = new DidDocument().fromPublicKey(genericSigningKey.publicKey)
```
Comment (katszwn) [inline ts/identity/index.ts]: I think we can remove this file
